Library/Homebrew/extend/pathname.rb

```diff
@@ -131,6 +131,12 @@ def write(content, *open_args)
     open("w", *open_args) { |f| f.write(content) }
   end
 
+  # Only appends to a file that is already created.
+  def write_append(content, *open_args)
+    raise "Cannot append file that doesn't exist: #{self}" unless exist?
+    open("a", *open_args) { |f| f.puts(content) }
```

Comment:
Taking @tdsmith's comment into account my view has shifted: Because it was named `write_append` I expected it to behave more like `write`, but naming it `append_lines` makes that less of an issue. The core of the issue is maybe that sometimes it's called with a single string, where it's desirable to have that being auto-terminated by a newline, and sometimes it's called with a heredoc that already contains a newline at the end. But from the method's point of view it's all just strings. (Sorry if these random thoughts that are somewhat useless. I wanted to share them anyway.)